### ISSUE TITLE
with_platform_config ignored for ImportExampleGen for KubeflowV2DagRunner

### ISSUE THREAD

**Issue by xivarri** [tensorflow/tfx#6146]
[This commit](https://github.com/tensorflow/tfx/commit/b15d592369d2db868f40185e689e9f6d43358dc6) added support for using `with_platform_config` with `KubeflowV2DagRunner` for running on Vertex, but because the logic only runs in `_build_container_spec` in `step_builder.py`, it is not run for `Importer`/`FileBasedExampleGen` nodes, so adding `with_platform_config` to e.g. `ImportExampleGen` is not reflected in the output json.

**Comment by singhniraj08** [comment]
@xivarri, Can you try using below code on `ImportExampleGen` or `FileBasedExampleGen` components to specify CPU and RAM on Vertex as shown in [similar issue](https://github.com/tensorflow/tfx/issues/3194#issuecomment-1573311503).

Thank you!

```
from kfp.pipeline_spec import pipeline_spec_pb2 as pipeline_pb2

my_component = MyComponent().with_platform_config(
        pipeline_pb2.PipelineDeploymentConfig.PipelineContainerSpec
        .ResourceSpec(cpu_limit=2.0, memory_limit=4.0))
```

**Comment by xivarri** [comment]
Yeah that's what I did - but this does not get reflected in the output json, I had to do this hack from that issue to manually modify the json: 
```
    for component_executor_spec in data["pipelineSpec"]["deploymentSpec"][
        "executors"
    ].values():
        if "resources" not in component_executor_spec["container"]:
            component_executor_spec["container"]["resources"] = {
                "cpuLimit": 32.0,
                "memoryLimit": 128.0,
            }
```
 
(I also could not limit this to only the importexamplegen's executor - if I did that vertex would still use e2-standard-4 for it - so set it for all of them)

**Comment by singhniraj08** [comment]
@xivarri, Thank you for sharing this insights. We will work on this internally and update the thread.

**Comment by Aditya-171** [comment]
Does kubeflow supports .json file .? from when .?
As per I know it only supports yaml, zip, tar.gz files .

How are you able to run pipeline json files in kubeflow?

@xivarri

**Comment by bharatjetti** [comment]
Hi @xivarri  @Aditya-171 Thank you so much for showing interest! We truly appreciate your enthusiasm and support.
We regret to inform you that we are not accepting new implementation requests or feature additions at this time. Please feel free to fork the repository and submit pull requests if you have any suggestions for enhancements, problem fixes, or ideas. We’d be more than happy to review and collaborate! If you would like to contribute , please follow our [CONTRIBUTION GUIDELINES](https://github.com/tensorflow/tfx/blob/master/CONTRIBUTING.md) here and submit a pull request.
Thank you again for your interest, and we look forward to seeing your contributions!

**Comment by bharatjetti** [comment]
Closing this due to inactivity. Please feel free to reopen and post your comments(if you still have queries on this). Thank you!
